src/earn/EarnEnterAmount.tsx

```diff
@@ -0,0 +1,626 @@
+  const tokenAmountRegex = new RegExp(
+    `^(?:\\d+[${decimalSeparator}]?\\d*|[${decimalSeparator}]\\d*|[${decimalSeparator}])$`
+  )
+  const parsedTokenAmount = useMemo(
+    () => parseInputAmount(tokenAmountInput, decimalSeparator),
+    [tokenAmountInput]
+  )
+  const parsedLocalAmount = useMemo(
+    () =>
+      parseInputAmount(
+        localAmountInput.replaceAll(groupingSeparator, '').replace(localCurrencySymbol, ''),
+        decimalSeparator
+      ),
+    [localAmountInput]
+  )
+
+  const tokenToLocal = useTokenToLocalAmount(parsedTokenAmount, token.tokenId)
+  const localToToken = useLocalToTokenAmount(parsedLocalAmount, token.tokenId)
+  const { tokenAmount, localAmount } = useMemo(() => {
+    if (enteredIn === 'token') {
+      setLocalAmountInput(
+        tokenToLocal && tokenToLocal.gt(0)
+          ? `${localCurrencySymbol}${tokenToLocal.toFormat(2)}` // automatically adds grouping separators
+          : ''
+      )
+      return {
+        tokenAmount: parsedTokenAmount,
+        localAmount: tokenToLocal,
+      }
+    } else {
+      setTokenAmountInput(
+        localToToken && localToToken.gt(0)
+          ? // no group separator for token amount, round to token.decimals and strip trailing zeros
+            localToToken
+              .toFormat(token.decimals, { decimalSeparator })
+              .replace(new RegExp(`[${decimalSeparator}]?0+$`), '')
+          : ''
+      )
+      return {
+        tokenAmount: localToToken,
+        localAmount: parsedLocalAmount,
+      }
+    }
+  }, [tokenAmountInput, localAmountInput, enteredIn, token])
+
+  const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, token.networkId))
+
+  useEffect(() => {
+    clearPreparedTransactions()
+
+    if (
+      !tokenAmount ||
+      tokenAmount.isLessThanOrEqualTo(0) ||
+      tokenAmount.isGreaterThan(token.balance)
+    ) {
+      return
+    }
+    const debouncedRefreshTransactions = setTimeout(() => {
+      return handleRefreshPreparedTransactions(tokenAmount, token, feeCurrencies)
+    }, FETCH_UPDATED_TRANSACTIONS_DEBOUNCE_TIME)
+    return () => clearTimeout(debouncedRefreshTransactions)
+  }, [tokenAmount, token])
+
+  const isAmountLessThanBalance = tokenAmount && tokenAmount.lte(token.balance)
+  const showNotEnoughBalanceForGasWarning =
+    isAmountLessThanBalance &&
+    prepareTransactionsResult &&
+    prepareTransactionsResult.type === 'not-enough-balance-for-gas'
+  const transactionIsPossible =
+    isAmountLessThanBalance &&
+    prepareTransactionsResult &&
+    prepareTransactionsResult.type === 'possible' &&
+    prepareTransactionsResult.transactions.length > 0
+
+  const disabled =
+    !!tokenAmount?.isZero() || (!!tokenAmount?.lte(token.balance) && !transactionIsPossible) // Should disable if the user enters 0 or has enough balance but the transaction is not possible, shouldn't disable if they enter an amount larger than their balance as they will go to add flow
```

Comment:
nit: this won't be readable in the editor, move the comment to one line above and break it up to multiple lines. would be nice if prettier can do this.